specifications/environment-management.org

```diff
@@ -0,0 +1,95 @@
+* Environment Management: Scope
+
+Specification of how Snapcraft manages local and remote build
+environments and their lifecycle.
+
+This includes locally and remotely provisioned containers, VMs and build
+services such as Launchpad.
+
+* Terminology
+
+** Snapcraft Build Environment Counter (build-count)
+
+Counter that increments every time Snapcraft creates a new build environment.
+Initialized to zero =0=.  Tracked in the environment-manager datastore.
+
+** Project Instance Identification (project-id)
+
+Unique identifier for project, derived from counter =build-count=. The counter
+will be atomically incremented when claiming an identifier.
+
+** Build Environment Instance Naming (project-instance-name)
+
+The build environment's user-facing name.
+
+Format: =snapcraft-<project-id>-<project-name>-<project-id>=
+
+** "Stale" Build Environment
+
+A build environment that is older than 7 days.
```

Comment:
7 days should be configurable and we should define where here

Reply:
I'm thinking we switched from daily to release images, maybe we shouldn't expire it unless the fingerprint changes?

Reply:
I'm going to abandon the stale environment concept for now.  We'll just improve our checks to make sure the environment is sane, as necessary.